lib/nerves_hub_link.ex

```diff
@@ -6,47 +6,47 @@ defmodule NervesHubLink do
   """
   @spec connected? :: boolean()
   def connected?() do
-    Socket.check_connection(:device)
+    Socket.check_connection(Socket, :device)
   end
 
   def console_connected?() do
-    Socket.check_connection(:console)
+    Socket.check_connection(Socket, :console)
   end
 
   @doc """
   Checks if the device has a socket connection with NervesHub
   """
   def socket_connected?() do
-    Socket.check_connection(:socket)
+    Socket.check_connection(Socket, :socket)
   end
 
   @doc """
   Return whether there's currently an active console session
   """
   @spec console_active?() :: boolean()
-  defdelegate console_active?, to: Socket
+  def console_active?, do: Socket.console_active?(Socket)
 
   @doc """
   Current status of the update manager
   """
   @spec status :: NervesHubLink.UpdateManager.State.status()
-  defdelegate status(), to: NervesHubLink.UpdateManager
+  def status(), do: NervesHubLink.UpdateManager.status(Socket)
 
   @doc """
   Restart the socket and device channel
   """
-  @spec reconnect() :: :ok
-  defdelegate reconnect(), to: Socket
+  @spec reconnect_socket() :: :ok
+  def reconnect_socket(), do: Socket.reconnect_socket(Socket)
```

Comment:
Is there a specific reason for this change? This could be considered breaking since `reconnect()` has been around a while

Reply:
yeah  `use Slipstream` imports a `reconnect/2` function. We could make `NervesHubLink.reconnect` stay the same while the socket module remains the new function.